src/modules/eventlisteners.ts

```diff
@@ -52,6 +52,9 @@ function updateEventListeners(oldVnode: VNode, vnode?: VNode): void {
 
   // optimization for reused immutable handlers
   if (oldOn === on) {
+    if (vnode) {
+      (vnode as any).listener = (oldVnode as any).listener;
```

Comment:
I think you're right, and there is a problem here, but this patch is not sufficient. Reusing listeners is tricky, because the listener has a backpointer to the corresponding vnode, and you will still be loosing that, which is covered by `listener.vnode = vnode;` (line 84).